src/Microsoft.VisualStudio.ProjectSystem.Managed/ProjectSystem/LoadedActiveConfiguredProjectsDataSource.cs

```diff
@@ -0,0 +1,62 @@
+﻿// Licensed to the .NET Foundation under one or more agreements. The .NET Foundation licenses this file to you under the MIT license. See the LICENSE.md file in the project root for more information.
+
+using System.Threading.Tasks.Dataflow;
+
+namespace Microsoft.VisualStudio.ProjectSystem;
+
+/// <summary>
+///     Force loads the active <see cref="ConfiguredProject"/> objects so that any configured project-level
+///     services, such as evaluation and build services, are started.
+/// </summary>
+[Export(typeof(ILoadedActiveConfiguredProjectDataSource))]
+internal class LoadedActiveConfiguredProjectsDataSource : ChainedProjectValueDataSourceBase<IEnumerable<ConfiguredProject>>, ILoadedActiveConfiguredProjectDataSource
```

Comment:
I don';t understand the difference between ActiveConfiguredProjectsLoader and this?

Reply:
I agree. We should not have double copied the code. What is the reason we revert the change and create a new class? What didn't work in the original change? I think we need make code change based on understanding the problem, instead of trying with slightly different code.  Basically, this code has similar problem like last change in the original place, it didn't follow dataflow patterns correctly. It need JoinUpstream blocks, and because the transform block has a complex function like this, it need run it inside a JTF.RunAsync to prevent deadlocks.